src/libraries/System.Private.CoreLib/src/System/HashCode.cs

```diff
@@ -319,6 +320,60 @@ public void AddBytes(ReadOnlySpan<byte> value)
             ref byte pos = ref MemoryMarshal.GetReference(value);
             ref byte end = ref Unsafe.Add(ref pos, value.Length);
 
+            if (Unsafe.ByteOffset(ref pos, ref end) < (sizeof(int) * 4))
```

Comment:
Does this lead to better codegen than just `if (value.Length < 4)`?

Reply:
Yes, the generated assembly is a bit smaller. I will change it.